scripts/setup_osx.sh

```diff
@@ -1,38 +1,248 @@
-#!/bin/bash
-# Make sure that pip is available
-hash brew 2>/dev/null || {
-  echo "Please install homebrew before continuing. You can use the" \
-    "following command to install:"
-  echo "/usr/bin/ruby -e \"\$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)\""
-  exit 0
+#!/usr/bin/env bash
+# This script installs garage on OS X distributions.
+#
+# NOTICE: To keep consistency across this script, scripts/setup_linux.sh and
+# docker/Dockerfile.ci, if there's any changes applied to this file, specially
+# regarding the installation of dependencies, apply those same changes to the
+# mentioned files.
+
+### START OF CODE GENERATED BY Argbash v2.6.1 one line above ###
+die()
+{
+  local _ret=$2
+  test -n "$_ret" || _ret=1
+  test "$_PRINT_HELP" = yes && print_help >&2
+  echo "$1" >&2
+  exit ${_ret}
 }
 
-hash conda 2>/dev/null || {
-  echo "Please install anaconda before continuing. You can download" \
-    "it at https://www.continuum.io/downloads. Please use the Python 2.7" \
-    " installer."
-  exit 0
+begins_with_short_option()
+{
+  local first_option all_short_options
+  all_short_options='h'
+  first_option="${1:0:1}"
+  test "$all_short_options" = "${all_short_options/$first_option/}" && \
+    return 1 || return 0
 }
 
 
-echo "Installing system dependencies"
-echo "You will probably be asked for your sudo password."
 
-brew install swig sdl sdl_image sdl_mixer sdl_ttf portmidi openmpi
+# THE DEFAULTS INITIALIZATION - POSITIONALS
+_positionals=()
+# THE DEFAULTS INITIALIZATION - OPTIONALS
+_arg_set_envvar="off"
 
-# Make sure that we're under the directory of the project
-cd "$(dirname "$0")/.."
+print_help ()
+{
+  printf '%s\n' "Installer of garage for Linux."
+  printf 'Usage: %s [--(no-)set-envvar] ' "$0"
+  printf '[-h|--help] <mjkey-path>\n' "$0"
+  printf '\t%s\n' "<mjkey-path>: Path of the MuJoCo key"
+  printf '\t%s' "--set-envvar,--no-set-envvar: Set environment variables "
+  printf '%s\n' "required by garage in .bash_profile (off by default)"
+  printf '\t%s\n' "-h,--help: Prints help"
+}
 
-echo "Setting up MuJoCo..."
-/bin/bash ./scripts/setup_mujoco.sh
-if [[ "${?}" -ne 0 ]]; then
-  echo -e "\e[0;31mError: MuJoCo couldn't be set up\e[0m"
-  exit 1
+parse_commandline ()
+{
+  while test $# -gt 0
+  do
+    _key="$1"
+    case "$_key" in
+      --no-set-envvar|--set-envvar)
+        _arg_set_envvar="on"
+        test "${1:0:5}" = "--no-" && _arg_set_envvar="off"
+        ;;
+      -h|--help)
+        print_help
+        exit 0
+        ;;
+      -h*)
+        print_help
+        exit 0
+        ;;
+      *)
+        _positionals+=("$1")
+        ;;
+    esac
+    shift
+  done
+}
+
+
+handle_passed_args_count ()
+{
+  _required_args_string="'mjkey-path'"
+  error_msg="$(echo "FATAL ERROR: Not enough positional arguments - we" \
+    "require exactly 1 (namely: $_required_args_string), but got only" \
+    "${#_positionals[@]}.")"
```

Comment:
IIRC MuJoCo recommends moving the mjkey.txt into the `.mujoco` folder, so maybe we can remind users that their `mjkey.txt` is located for example in `./scripts/setup_osx.sh ~/.mujoco/mjkey.txt`. At least for me, I had to think for a while about where I last stored my mjkey.txt after seeing this error.

Reply:
I don't understand this request very well. This script asks for the path where mjkey.txt is so it can be moved into ~/.mujco/mjkey.txt as shown [here](https://github.com/rlworkgroup/garage/blob/c580437b33f14fd905efecf00bb577637f85a16d/scripts/setup_osx.sh#L223).
For example, if the key is downloaded in ~/Downloads/mjkey.txt, then it's copied into ~/.mujoco/mjkey.txt.
Do you mean printing a help message or something like that?

Reply:
Ignore what I said previously, I didn't see that line.

Reply:
I agree with Angel that it's fine.